cancermuts/datasources.py

```diff
@@ -1161,6 +1193,107 @@ def _get_revel_from_gc(self, mutation, gc):
+            except KeyError:
+                self.log.warning("RevelLocal doesn't support metadata type %s" % md)
+
+        transcript_id = sequence.aliases.get('ensembl_transcript_id')
+        if not transcript_id:
+            self.log.warning(f"No Ensembl transcript ID available for {sequence}")
+            return
+
+        mutations = []
+        for pos in sequence.positions:
+            for mut in pos.mutations:
+                gms = mut.metadata.get('genomic_mutations', [])
+                if not gms:
+                    self.log.debug(f"[REVEL] No genomic mutation available for {mut}; skipping.")
+                    continue
+                mutations.append((mut, gms, transcript_id))
+
+        self._get_revel(mutations)
+
+    def _get_revel(self, mutation_entries):
+        df = self._revel_df
+
+        for mutation, gms, transcript_id in mutation_entries:
+            mutation.metadata['revel_score'] = []
+
+            aaref = mutation.sequence_position.wt_residue_type
+            aaalt = mutation.mutated_residue_type
+
+            for gm in gms:
+                if not hasattr(gm, 'genome_build') or not hasattr(gm, 'chr') or not hasattr(gm, 'get_coord') or not hasattr(gm, 'ref') or not hasattr(gm, 'alt'):
+                    self.log.warning(f"[REVEL] Skipping invalid genomic mutation object: {gm}")
+                    continue
+
+                genome_version = gm.genome_build
+                chrom = str(gm.chr)
+                pos = str(gm.get_coord())
+                ref = gm.ref
+                alt = gm.alt
+
+                coord_col = "grch38_pos" if genome_version == "hg38" else "hg19_pos" if genome_version == "hg19" else None
+                if coord_col is None:
+                    self.log.warning(f"[REVEL] Unknown genome version '{genome_version}' for {mutation}")
+                    continue
+
+                df_filtered = df[
+                    (df["chr"].astype(str) == chrom) &
+                    (df[coord_col].astype(str) == pos)
+                ]
+
+                if df_filtered.empty:
+                    self.log.debug(f"[REVEL] No position match for {mutation} at chr {chrom}, pos {pos}")
+                    continue
+
+                for _, row in df_filtered.iterrows():
+                    if (
+                        row["ref"] != ref or
+                        row["alt"] != alt or
+                        row["aaref"] != aaref or
+                        row["aaalt"] != aaalt
+                    ):
+                        continue
+
+                    if transcript_id not in row["Ensembl_transcriptid"].split(";"):
+                        continue
+
+                    score_str = row["REVEL"]
+                    if score_str in [".", "NA"]:
+                        continue
+
+                    try:
+                        score = float(score_str)
+                        mutation.metadata['revel_score'].append(DbnsfpRevel(source=self, score=score))
```

Comment:
we should change the name of the `DbnsfRevel` class to `Revel`, because in this case it's not from DBNSF